Signum.Engine.Extensions/Authorization/ResetPasswordRequestLogic.cs

```diff
@@ -20,37 +22,60 @@ public static class ResetPasswordRequestLogic
     {
         public static void Start(SchemaBuilder sb)
         {
-            if (sb.NotDefined(MethodInfo.GetCurrentMethod()))
-            {
-                sb.Include<ResetPasswordRequestEntity>()
-                    .WithQuery(() => e => new
-                    {
-                        Entity = e,
-                        e.Id,
-                        e.RequestDate,
-                        e.Code,
-                        e.User,
-                        e.User.Email
-                    });
-
-                EmailLogic.AssertStarted(sb);
+            if (!sb.NotDefined(MethodInfo.GetCurrentMethod()))
+                return;
 
-                EmailModelLogic.RegisterEmailModel<ResetPasswordRequestEmail>(() => new EmailTemplateEntity
+            sb.Include<ResetPasswordRequestEntity>()
+                .WithQuery(() => e => new
                 {
-                    DisableAuthorization = true,
-                    Messages = CultureInfoLogic.ForEachCulture(culture => new EmailTemplateMessageEmbedded(culture)
-                    {
-                        Text = "<p>{0}</p>".FormatWith(AuthEmailMessage.YouRecentlyRequestedANewPassword.NiceToString()) +
-                            "<p>{0} @[User.UserName]</p>".FormatWith(AuthEmailMessage.YourUsernameIs.NiceToString()) +
-                            "<p>{0}</p>".FormatWith(AuthEmailMessage.YouCanResetYourPasswordByFollowingTheLinkBelow.NiceToString()) +
-                            "<p><a href=\"@[m:Url]\">@[m:Url]</a></p>",
-                        Subject = AuthEmailMessage.ResetPasswordRequestSubject.NiceToString()
-                    }).ToMList()
+                    Entity = e,
+                    e.Id,
+                    e.RequestDate,
+                    e.Code,
+                    e.User,
+                    e.User.Email
                 });
 
+            EmailLogic.AssertStarted(sb);
+
+            SimpleTaskLogic.Register(ResetPasswordRequestTask.Timeout,
+                (ScheduledTaskContext ctx) =>
+                {
+                    Database.Query<ResetPasswordRequestEntity>()
+                        .Where(r => r.RequestDate.AddHours(24) < TimeZoneManager.Now)
+                        .UnsafeUpdate()
+                        .Set(e => e.Lapsed, e => true)
+                        .Execute();
 
+                    return null;
+                }
+            );
 

```

Comment:
I understand that you want to discard old ResetPasswordRequests, but an scheduled tasks daily looks like an overkill. 
What about renaming Lapsed to Used and add a new:
`bool IsValid => As.Expression(()=> !Used && TimeZoneManager.Now < r.RequestDate.AddHours(24))`

Reply:
Removed task in favor of IsValid expression. See: [a5a2ac33e](https://github.com/signumsoftware/extensions/pull/462/commits/a5a2ac33e617734c1a7ee4219405b6ec46b57dba)